library/Icinga/Common/HealthBadgeTrait.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Icinga\Common;
+
+use Icinga\Application\Hook\HealthHook;
+use ipl\Web\Widget\StateBadge;
+
+trait HealthBadgeTrait
+{
+    private $STATE_OK = 'ok';
+    private $STATE_CRITICAL = 'critical';
+    private $STATE_WARNING = 'warning';
+    private $STATE_PENDING = 'pending';
+    private $STATE_UNKNOWN = 'unknown';
+
+    protected $state;
+
+    protected function createHealthBadge()
+    {
+        $stateBadge = null;
+        if ($this->getHealthCount() > 0) {
+            $stateBadge = new StateBadge($this->getHealthCount(), $this->state);
+            $stateBadge->addAttributes(['class' => 'disabled']);
+        }
+
+        return $stateBadge;
+    }
+
+    protected function getHealthCount()
+    {
+        $count = 0;
+        $title = null;
+        $worstState = null;
+        foreach (HealthHook::collectHealthData()->select() as $result) {
+            if ($worstState === null || $result->state > $worstState) {
+                $worstState = $result->state;
+                $title = $result->message;
+                $count = 1;
+            } elseif ($worstState === $result->state) {
+                $count++;
+            }
+        }
+
+        switch ($worstState) {
+            case HealthHook::STATE_OK:
+                $count = 0;
+                break;
+            case HealthHook::STATE_WARNING:
+                $this->state = $this->STATE_WARNING;
+                break;
+            case HealthHook::STATE_CRITICAL:
+                $this->state = $this->STATE_CRITICAL;
+                break;
+            case HealthHook::STATE_UNKNOWN:
+                $this->state = $this->STATE_UNKNOWN;
+                break;
+        }
+
+        $this->title = $title;
```

Comment:
Didn't notice that in #4651.. but why is this unused? And dynamically initialized?

Couldn't this be used as title for the state badge?

Reply:
✅